File: server/src/Timers/SpawnAmmoTimer.cs
using SpacetimeDB;
using Game = pillz.server.Reducers.Game;

namespace pillz.server.Timers;

[Table(Scheduled = nameof(Game.SpawnPrimaryAmmo), ScheduledAt = nameof(ScheduledAt))]
public partial struct SpawnAmmoTimer
{
    [PrimaryKey, AutoInc]
    public ulong ScheduledId;
    public ScheduleAt ScheduledAt;
}

File: server-rust/src/timers/spawn_ammo_timer.rs
use spacetimedb::ScheduleAt;
use crate::reducers::game::spawn_ammo;

#[spacetimedb::table(name = spawn_ammo_timer, scheduled(spawn_ammo))]
pub struct SpawnAmmoTimer {
    #[primary_key]
    #[auto_inc]
    pub scheduled_id: u64,

    pub scheduled_at: ScheduleAt,
}
